### PR TITLE
3852.json welcome page

### PR DESCRIPTION
**Pull request by meejah**
fixes #3852

### CODE REVIEW

File: src/allmydata/test/web/test_root.py [hunk 1]
```diff
@@ -138,3 +149,76 @@ def __init__(self):
 
         self.assertThat(item.slotData.get("version"), Equals(""))
         self.assertThat(item.slotData.get("nickname"), Equals(""))
+
+
+class RenderRoot(AsyncTestCase):
+
+    @inlineCallbacks
+    def test_root_json(self):
+        """
```
Comment (itamarst) [inline src/allmydata/test/web/test_root.py:158]: Can you add a docstring?

File: newsfragments/3852.minor [hunk 2]
```diff
@@ -0,0 +1 @@
+
```
Comment (itamarst) [inline newsfragments/3852.minor:1]: Now that Python 3 is publicly available, bug fixes to Python 3 support are real bugfixes. So this should be `.bugfix` with appropriate summary.

File: src/allmydata/test/web/test_root.py [hunk 3]
```diff
@@ -138,3 +149,76 @@ def __init__(self):
 
         self.assertThat(item.slotData.get("version"), Equals(""))
         self.assertThat(item.slotData.get("nickname"), Equals(""))
+
+
+class RenderRoot(AsyncTestCase):
+
+    @inlineCallbacks
+    def test_root_json(self):
```
Comment (itamarst) [inline src/allmydata/test/web/test_root.py:157]: Does this actually reproduce the problem when run without the fix? Seems like you would need to have two servers, one with availability, one without. If I'm correct, you should do the "make sure the test actually fail when fix is commented out" dance.

**Review by itamarst — APPROVED** [review]
Please address review comments, and then merge once #1169 allows tests to pass again.

File: src/allmydata/test/web/test_root.py (hunk 3 above)
Comment (meejah) [reply src/allmydata/test/web/test_root.py:157]: Thanks, I'll double-check (I did confirm this branch works with magic-folders but don't recall if I confirmed the utest ;)

Comment (meejah) [reply src/allmydata/test/web/test_root.py:157]: indeed it did not, thanks!

**Pull request merged**